cmd/root.go

```diff
@@ -196,6 +196,10 @@ func rootCmdPersistentFlagSet(gs *state.GlobalState) *pflag.FlagSet {
 		"enable profiling (pprof) endpoints, k6's REST API should be enabled as well",
 	)
 
+	flags.StringVar(&gs.Flags.TracesOutput, "traces-output", gs.Flags.TracesOutput,
+		"set the output for k6 traces, possible values are none,otel[=host:port]")
+	flags.Lookup("traces-output").DefValue = gs.DefaultFlags.TracesOutput
+
```

Comment:
This should be in the [runtime options](https://github.com/grafana/k6/blob/master/cmd/runtime_options.go) and only be parsed and applied on `k6 run` which I think it does automatically if it is there :thinking:

Reply:
I believe there is an issue related to moving the _traces output_ flag to the runtime options. The runtime options are parsed in the call to [`loadAndConfigureTest`](https://github.com/grafana/k6/blob/85fdca1c857bae7609b7efc248eab98b6ad0a888/cmd/run.go#L114) from `run` command execution, which is the same function that sets the `TracerProvider` in `TestPreInitState` from global state, which means that at that time the `TracerProvider` must have been initialized.

So what do you think about moving the _traces output_ option to the runtime options and then adding a new method to just parse this option, such as:
<details>
<summary>code</summary>

```go
// TODO: Merge this into getRuntimeOptions when ...
func getTracesOutputOption(flags *pflag.FlagSet, environment map[string]string) null.String {
	to := getNullString(flags, "traces-output")
	if envVar, ok := environment["K6_TRACES_OUTPUT"]; ok {
		if !to.Valid {
			to = null.StringFrom(envVar)
		}
	}

	return to
}
```
</details>

Then this will be called from `setupTracerProvider` method. We can still add the `TracesOutput` field to `lib.RuntimeOptions` and [call](https://github.com/grafana/k6/blob/85fdca1c857bae7609b7efc248eab98b6ad0a888/cmd/runtime_options.go#L70) this `getTracesOutputOption` method from `getRuntimeOptions` to parse it, so we can add the corresponding tests for it, and in the future just remove this method call and merge the implementation in `getRuntimeOptions`.

Not the nicest. but might be the best option until the `Runner` interface is modified to accept the `TracerProvider` as input so this can be initialized later and use the parsed configuration from `loadAndConfigureTest` call.

Reply:
I am not certain why you can't just move the tracer initalization *after* the call to `loadAndConfigureTest` at this point ?

Reply:
By the looks of it you can:
1. move the whole configuration in the runtime options.
2. remove anything about it from the global state
3. use the TestPreinitstate to both get the runtime optiosn and set it's tracer output in the setupTracerProvider

Reply:
That seems a little bit "obscure" to me, as we are modifying the `TracerProvider` by reference, when this one has already been propagated along with the `TestPreInitState` to the [`Runner`](https://github.com/grafana/k6/blob/ef550a555115fe8397a0a099055a2b750f4fef4a/js/runner.go#L95) in the call to `loadAndConfigureTest`.
Done in [5a24571](https://github.com/grafana/k6/pull/3445/commits/5a24571f66f9dadefb901a36d1c029bee6bf2534)..[ef550a5](https://github.com/grafana/k6/pull/3445/commits/ef550a555115fe8397a0a099055a2b750f4fef4a).